docs/date.md

```diff
@@ -1,96 +1,398 @@
+date = Temporal.Date.from('2006-08-24T15:43:27Z');  // => 2006-08-24
+date = Temporal.Date.from('2006-08-24T15:43:27+01:00[Europe/Brussels]');
+  // => 2006-08-24
+date === Temporal.Date.from(date)  // => true
+
+date = Temporal.Date.from({year: 2006, month: 8, day: 24});  // => 2006-08-24
+date = Temporal.Date.from(Temporal.DateTime.from('2006-08-24T15:43:27'));
+  // => same as above; Temporal.DateTime has year, month, and day properties
+```
 
-Allows for easier comparison of `Date` objects, returns:
+### Temporal.Date.**compare**(_one_: Temporal.Date, _two_: Temporal.Date) : number
 
-- `-1` if the first object represents a lower value than the second
-- `0` if the 2 objects represent the same value
-- `1` if the first object represents a higher value than the second
+**Parameters:**
+- `one` (`Temporal.Date`): First date to compare.
+- `two` (`Temporal.Date`): Second date to compare.
+
+**Returns:** &minus;1, 0, or 1.
+
+Compares two `Temporal.Date` objects.
+Returns an integer indicating whether `one` comes before or after or is equal to `two`.
+- &minus;1 if `one` comes before `two`;
+- 0 if `one` and `two` are the same;
+- 1 if `one` comes after `two`.
+
+This function can be used to sort arrays of `Temporal.Date` objects.
+For example:
+```javascript
+one = Temporal.Date.from('2006-08-24');
+two = Temporal.Date.from('2015-07-14');
+three = Temporal.Date.from('1930-02-18');
+sorted = [one, two, three].sort(Temporal.Date.compare);
+sorted.join(' ');  // => 1930-02-18 2006-08-24 2015-07-14
+```
 
 ## Properties
 
 ### date.**year** : number
 
-Returns the year this `Date` represents
-
 ### date.**month** : number
 
-Returns the month this `Date` represents
-
 ### date.**day** : number
 
-Returns the day this `Date` represents
+The above read-only properties allow accessing each component of the date individually.
+
+Usage examples:
+```javascript
+date = Temporal.Date.from('2006-08-24');
+date.year   // => 2006
+date.month  // => 8
+date.day    // => 24
+```
 
 ### date.**dayOfWeek** : number
 
-Returns the day of the week this `Date` represents
+The `dayOfWeek` read-only property gives the weekday number that the date falls on.
+The weekday number is a value between 1 and 7, inclusive: Monday is 1, and Sunday is 7.
 
-```js
-const date = new Temporal.Date(2019, 11, 18); // Monday
-date.dayOfWeek; // 1
+Usage example:
+```javascript
+date = Temporal.Date.from('2006-08-24');
+['MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT', 'SUN'][date.dayOfWeek - 1]  // => THU
+```
+
+### date.**dayOfYear** : number
+
+The `dayOfYear` read-only property gives the ordinal day of the year that the date falls on.
+This is a value between 1 and 365, or 366 in a leap year.
+
+Usage example:
+```javascript
+date = Temporal.Date.from('2006-08-24');
+// Print date in ISO ordinal date format
+year = `${date.year}`.padStart(4, '0');
+day = `${date.dayOfYear}`.padStart(3, '0');
+`${year}-${day}`  // => 2006-236
```

Comment:
**Here and elsewhere:**  I see what you're trying to do, but please, let's not put string formatting in the docs.  We don't want to encourage that behavior.  People should use to[ISO]String or toLocaleString.  Just do something like `console.log(date.year, date.dayOfYear)` without the string formatting.

Reply:
Point taken, although that does make these examples less exciting; and people can't actually get these date formats from toString or toLocaleString.